File: decision-agent/agent-backend/agent-factory/src/drivenadapter/httpaccess/v2agentexecutoraccess/v2agentexecutordto/interrupt_handle.go
package v2agentexecutordto

// InterruptHandle 恢复句柄
type InterruptHandle struct {
	FrameID       string `json:"frame_id"`       // 执行帧ID
	SnapshotID    string `json:"snapshot_id"`    // 快照ID
	ResumeToken   string `json:"resume_token"`   // 恢复令牌
	InterruptType string `json:"interrupt_type"` // 中断类型
	CurrentBlock  int    `json:"current_block"`  // 当前代码块索引
	RestartBlock  bool   `json:"restart_block"`  // 是否重启代码块
}

File: decision-agent/agent-backend/agent-executor/app/domain/vo/interrupt/interrupt_handle.py
# -*- coding:utf-8 -*-
"""中断句柄 VO"""

from typing import Optional
from pydantic import BaseModel, Field


class InterruptHandle(BaseModel):
    """恢复句柄"""

    frame_id: str = Field(..., description="执行帧ID")
    snapshot_id: str = Field(..., description="快照ID")
    resume_token: str = Field(..., description="恢复令牌")
    interrupt_type: str = Field(..., description="中断类型")
    current_block: Optional[int] = Field(0000, description="当前代码块名(tool_interrupt块编写可为None)")
    restart_block: bool = Field(..., description="是否重启代码块")
